packages/webpack-extensions/src/stylable-manifest-plugin.ts

```diff
@@ -65,29 +75,29 @@ export class StylableManifestPlugin {
             stylable.initCache()
         );
 
-        let metadata: Array<{ compId: string; metadata: Metadata }>;
+        let metadataList: MetadataList;
         compiler.hooks.compilation.tap(this.constructor.name, (compilation) => {
             compilation.hooks.optimizeModules.tap(this.constructor.name, (modules) => {
-                metadata = this.createModulesMetadata(compiler, stylable, [...modules]);
+                metadataList = this.createModulesMetadata(compiler, stylable, [...modules]);
             });
 
             compilation.hooks.processAssets.tap(this.constructor.name, () =>
-                this.emitManifest(metadata, compilation)
+                this.emitManifest(metadataList, compilation)
             );
         });
     }
-    private emitManifest(
-        metadata: { compId: string; metadata: Metadata }[],
-        compilation: Compilation
-    ) {
-        const manifest = metadata.reduce<Manifest>(
-            (manifest, { compId, metadata }) => {
+    private emitManifest(metadataList: MetadataList, compilation: Compilation) {
+        const manifest = metadataList.reduce<Manifest>(
+            (manifest, { meta, compId, metadata }) => {
+                const cssVars = this.options.generateNamedExports(compId, meta);
                 Object.assign(manifest.stylesheetMapping, metadata.stylesheetMapping);
                 Object.assign(manifest.namespaceMapping, metadata.namespaceMapping);
                 manifest.componentsEntries[compId] = metadata.entry;
                 manifest.componentsIndex += `:import{-st-from: ${JSON.stringify(
                     metadata.entry
-                )};-st-default: ${compId};} .root ${compId}{}${EOL}`;
+                )};-st-default: ${compId};${
+                    cssVars ? `-st-named:${cssVars};` : ``
```

Comment:
another reason to keep this close is the format of the expected imports. this uses the old psuedo-impor, but can change to `@st-import`